### PR TITLE
ft(UI): Add categories card in details panel and new criterias to filter on them

### PR DESCRIPTION
**Pull request by Jordan91**
## Description

Add categories card in details panel add posibility of filter on them by selected Criterias in the Filter

https://www.loom.com/share/8e2bccb9c26946d6ae460e40af569f9b

## Type of change

- [x] New functionality (non-breaking change)

## Target serie

- [x] 22.10.x (master)

<h2> How this pull request can be tested ? </h2>

Add service and host category in configuration , then set a host or service with one of them. 
Export and reload configuration
Go to the resources status menu and start to editing filter, to choose which category you want to display.

In details panel you can see a new card with the category set on the resources who looks like the groups's card.

### CODE REVIEW

File: www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx [hunk 1]
```diff
@@ -185,11 +188,38 @@ const getDetailCardLines = ({
...
+      isCustomCard: true,
+      line: (
+        <DetailsChips
+          getType={(): CriteriaNames =>
+            equals(details?.type, ResourceType.host)
+              ? CriteriaNames.hostCategories
+              : CriteriaNames.serviceCategories
+          }
+          groups={details?.categories}
+          title={labelCategory}
```
Comment (bdauria) [inline www/front_src/src/Resources/Details/tabs/Details/DetailsCard/cards.tsx:216]: Should be plural

File: www/front_src/src/Resources/Filter/Criterias/searchQueryLanguage/index.test.ts [hunk 2]
```diff
@@ -126,6 +140,16 @@ describe(getAutocompleteSuggestions, () => {
       expectedResult: [],
       inputSearch: 'host_group:',
     },
+    {
+      cursorPosition: 18,
+      expectedResult: [],
+      inputSearch: 'service_categorie:',
```
Comment (bdauria) [inline www/front_src/src/Resources/Filter/Criterias/searchQueryLanguage/index.test.ts:146]:
```suggestion
      inputSearch: 'service_category:',
```

File: www/front_src/src/Resources/Filter/Criterias/searchQueryLanguage/index.test.ts [hunk 3]
```diff
@@ -126,6 +140,16 @@ describe(getAutocompleteSuggestions, () => {
...
+    },
+    {
+      cursorPosition: 15,
+      expectedResult: [],
+      inputSearch: 'host_categorie:',
```
Comment (bdauria) [inline www/front_src/src/Resources/Filter/Criterias/searchQueryLanguage/index.test.ts:151]:
```suggestion
      inputSearch: 'host_category:',
```

File: www/front_src/src/Resources/Filter/index.test.tsx [hunk 4]
```diff
@@ -716,6 +774,24 @@ describe(Filter, () => {
...
+
+      await waitFor(() => expect(mockedAxios.get).toHaveBeenCalled());
+
+      const linuxServerCategory = await findByText(linuxHostCategory.name);
+
+      expect(linuxServerCategory).toBeInTheDocument();
+
+      act(() => {
+        fireEvent.click(getByText(labelServiceCategory));
+      });
```
Comment (bdauria) [inline www/front_src/src/Resources/Filter/index.test.tsx:788]:
```suggestion
      });
      
```

File: www/front_src/src/Resources/translatedLabels.ts [hunk 5]
```diff
@@ -249,3 +249,6 @@ export const labelNoContactIsConfiguredForThisResource =
   'No contacts are configured for this resource';
 export const labelNoContactGroupsIsConfiguredForThisResource =
   'No contact groups are configured for this resource';
+export const labelCategory = 'Categories';
```
Comment (bdauria) [inline www/front_src/src/Resources/translatedLabels.ts:252]: There's a incoherence here: do you want the label to be plural or singular? if plural is the answer, then the name should be labelCategories

Comment (Jordan91) [reply www/front_src/src/Resources/translatedLabels.ts:252]: yes :/ it's fixed

**Pull request closed without merging**